tests/test_lime_survey.py

```diff
@@ -214,5 +214,126 @@ def test_lime_system_info_dtypes(self):
         )
 
 
+class TestLimeSurveyGetLabels(unittest.TestCase):
```

Comment:
General comments for tests:
* Please, make tests more specific. Provide specific question names and specific outputs. Like, for example, `self.assetEqual(survey.get_label("B6_SQ001"), 'More time needed to complete PhD project')`. This will make it easier to locate the problem if a test fails.
* Test single column names (like "B6_SQ001") as well.

Reply:
I was under the impression that it's best practice to not hard code these names, and things should be kept general and not rely on this specific survey structure. If one were to run the tests with hard coded question names and expected answers as raw strings on a different structure xml, they would all fail no? Is the principle not application to designing tests?

Reply:
Sorry, for the late reply.

> I was under the impression that it's best practice to not hard code these names, and things should be kept general and not rely on this specific survey structure.

That is true for the main code, not for the tests. For tests, it is kinda opposite: specific input -> specific (expected) output. The reason is that for the main code it is better to be as flexible as possible, so, for example, in the case of surveys more different surveys can be processed. But for the unit tests, the main idea is to "catch" a problem, so it is better to have it as specific as possible, so if it fails we know what exactly went not as expected. 

> If one were to run the tests with hard coded question names and expected answers as raw strings on a different structure xml, they would all fail no?

Yes. But that is the point. Inputs for tests are not supposed to be modified without a particular reason.